lib/dry/types/hash/schema_builder.rb

```diff
@@ -4,15 +4,16 @@
 module Dry
   module Types
     class Hash < Definition
+      # A bulder for legacy schemas
+      # @api private
       class SchemaBuilder
-        def initialize
-          @nil_to_undefined = -> v { v.nil? ? Undefined : v }
-          @omittable_keys = %i(schema weak symbolized).freeze
-          @permissive = %i(schema permissive weak symbolized).freeze
-
-          freeze
-        end
+        NIL_TO_UNDEFINED = -> v { v.nil? ? Undefined : v }
+        OMITTABLE_KEYS = %i(schema weak symbolized).freeze
```

Comment:
could we rename this to `OPTIONAL_KEYS`? it'd be more consistent with dry-v's `required/optional`

Reply:
@solnic I don't have any preference about naming but this comes from `meta[:omittable]` where it means whether a key is optional. The problem is that this information is stored inside the type itself meaning if we change it from `:omittable` to `:optional` then we end up with something like

```ruby
age: Dry::Types['strict.int'].optional.meta(optional: true)
```
when we want have a key that can be omitted _and_ can have `nil` value. This looks rather confusing, don't you think? We don't have a separate place for key meta and I'd not go this way, yet.